File: prob12.py
#!/usr/bin/env python2.7

end = 500
I = 0
number = 0
while True:
    I += 1
    number += I
    number_of_divisors = 0
    j = 0
    stop_at = number
    while True:
        j += 1
        if j >= stop_at:
            break
        if number%j == 0:
            number_of_divisors += 2
            stop_at = number / j
    if number_of_divisors > end:
        print "First triangle to have more than",end,"divisors is",number
        break

File: prob12.f90
Program problem12
nend = 500
I = 0
num = 0
Do
  I = I + 1
  num = num+I
  ndiv = 0
  j = 0
  nstop = num
  Do
    j = j+1
    If ( j >= nstop ) Exit
    If ( mod(num,j) == 0 ) Then
      ndiv = ndiv+2
      nstop = num/j
    End If
  End Do
  !Print *, num,'has',ndiv,'divisors'
  If ( ndiv > nend ) Then
    Print *, 'First triangle to have more than',nend,'divisors is',num
    Exit
  End If
End Do
End Program problem12
